### PR TITLE
Cleanup warnings related to raw type List

### PR DESCRIPTION
**Pull request by BradWalker**
Cleanup warnings related to unchecked call to add as a member of raw type List..

Here is an example.

   [repeat] /home/bwalker/src/netbeans/ide/editor/src/org/netbeans/modules/editor/NbEditorKit.java:628: warning: [unchecked] unchecked call to add(E) as a member of the raw type List
   [repeat]                     objects.add(null);
   [repeat]                                ^
   [repeat]   where E is a type-variable:
   [repeat]     E extends Object declared in interface List

There are some remaining warnings. They are in generated code and therefore difficult to fix.

---
**^Add meaningful description above**

By opening a pull request you confirm that, unless explicitly stated otherwise, the changes -

 - are all your own work, and you have the right to contribute them.
 - are contributed solely under the terms and conditions of the Apache License 2.0 (see section 5 of the license for more information).

Please make sure (eg. `git log`) that all commits have a valid name and email address for you in the Author field.

If you're a first time contributor, see the Contributing guidelines for more information.

### CODE REVIEW

**Comment by mbien** [comment]
hmm. sig tests are failing but you didn't really change anything in LSP code.

```
Class org.netbeans.modules.java.lsp.server.ui.UIContext

  "E1.2 - API type removed" : method protected java.util.concurrent.CompletableFuture org.netbeans.modules.java.lsp.server.ui.UIContext.showQuickPick(org.netbeans.modules.java.lsp.server.protocol.ShowQuickPickParams)

  "E1.2 - API type removed" : method protected java.util.concurrent.CompletableFuture org.netbeans.modules.java.lsp.server.ui.UIContext.showInputBox(org.netbeans.modules.java.lsp.server.protocol.ShowInputBoxParams)

```

**Comment by matthiasblaesing** [comment]
@BradWalker could you please rebase this onto current master and see if that fixes the issue, so that we can get this in? These things don't get better with time.

File: enterprise/websvc.jaxwsmodel/src/org/netbeans/modules/websvc/api/jaxws/wsdlmodel/WsdlPort.java [hunk 1]
```diff
@@ -45,7 +47,7 @@ public class WsdlPort implements WSPort {
     }
     
     public List<WsdlOperation> getOperations() {
-        List wsdlOperations = new ArrayList();
+        List<WsdlOperation> wsdlOperations = new ArrayList<>();
```
Comment (lbownik) [inline enterprise/websvc.jaxwsmodel/src/org/netbeans/modules/websvc/api/jaxws/wsdlmodel/WsdlPort.java:50]: How about rewriting this funtion to

public List<WsdlOperation> getOperations() {
  if(port == null) {
      return emptyList();
  } else {
     return port.getOperations().stream().map(WdslOperation::new).collect(toList());
}

File: enterprise/websvc.jaxwsmodel/src/org/netbeans/modules/websvc/api/jaxws/wsdlmodel/WsdlService.java [hunk 2]
```diff
@@ -49,7 +51,7 @@ public WsdlService() {
     }
     
     public List<WsdlPort> getPorts() {
-        List wsdlPorts = new ArrayList();
+        List<WsdlPort> wsdlPorts = new ArrayList<>();
```
Comment (lbownik) [inline enterprise/websvc.jaxwsmodel/src/org/netbeans/modules/websvc/api/jaxws/wsdlmodel/WsdlService.java:54]: ho about rewriting this funtion into

public List<WsdlPort> getPorts() {
        
        if (service==null) {
            return emptyList();
        } else {
            return service.getPorts().stream().map(WsdlPort::new).collect(toList());
        }
}

File: enterprise/web.struts/src/org/netbeans/modules/web/struts/StrutsConfigUtilities.java [hunk 3]
```diff
@@ -304,7 +304,7 @@ public static String[] getConfigFiles(FileObject dd){
             Servlet servlet = getActionServlet(dd);
             if (servlet!=null) {
                 InitParam[] params = servlet.getInitParam();
```
Comment (lbownik) [inline enterprise/web.struts/src/org/netbeans/modules/web/struts/StrutsConfigUtilities.java:306]: How about rewriting it into

return Stream.of(servlet.getInitParam()).
   filter(p -> p.getParamName() != null).
   filter(p -> p.getParamName().startsWith(DEFAULT_MODULE_NAME)).
   flatMap(p -> Stream.of(p.getParamValue().split(",")).
   collect(toList())).
   toArray(new String[]{});

**Comment by BradWalker** [comment]
In process as we speak..

File: enterprise/web.struts/src/org/netbeans/modules/web/struts/StrutsConfigUtilities.java [hunk 4]
```diff
@@ -138,7 +138,7 @@ private static void addActions(List<Action> list, StrutsConfig sConfig) {
             mappings = sConfig.getActionMappings();
         }
         if (mappings==null) return;
-        Action [] actions = mappings.getAction();
+        Action[] actions = mappings.getAction();
```
Comment (matthiasblaesing) [inline enterprise/web.struts/src/org/netbeans/modules/web/struts/StrutsConfigUtilities.java:141]: Whitespace only change

File: enterprise/websvc.jaxwsmodel/src/org/netbeans/modules/websvc/api/jaxws/wsdlmodel/WsdlPort.java [hunk 5]
```diff
@@ -19,11 +19,13 @@
...
 
+import java.util.ArrayList;
+import java.util.List;
+
 import com.sun.tools.ws.processor.model.Operation;
 import com.sun.tools.ws.processor.model.Port;
 import com.sun.tools.ws.wsdl.document.soap.SOAPStyle;
-import java.util.ArrayList;
-import java.util.List;
+
```
Comment (matthiasblaesing) [inline enterprise/websvc.jaxwsmodel/src/org/netbeans/modules/websvc/api/jaxws/wsdlmodel/WsdlPort.java:28]: Order only change

File: java/debugger.jpda.ui/src/org/netbeans/modules/debugger/jpda/ui/FixedWatchesManager.java [hunk 6]
```diff
@@ -265,7 +265,7 @@ public void performDefaultAction (
     @Override
     public Action[] getActions (NodeActionsProvider original, Object node) 
     throws UnknownTypeException {
-        Action [] actions = original.getActions (node);
+        Action[] actions = original.getActions (node);
```
Comment (matthiasblaesing) [inline java/debugger.jpda.ui/src/org/netbeans/modules/debugger/jpda/ui/FixedWatchesManager.java:268]: Whitespace only change

**Comment by mbien** [comment]
please make sure the commit has a useful message prior to merge.

File: enterprise/web.struts/src/org/netbeans/modules/web/struts/StrutsConfigUtilities.java (hunk 4 above)
Comment (BradWalker) [reply enterprise/web.struts/src/org/netbeans/modules/web/struts/StrutsConfigUtilities.java:141]: yes..

File: enterprise/websvc.jaxwsmodel/src/org/netbeans/modules/websvc/api/jaxws/wsdlmodel/WsdlPort.java (hunk 5 above)
Comment (BradWalker) [reply enterprise/websvc.jaxwsmodel/src/org/netbeans/modules/websvc/api/jaxws/wsdlmodel/WsdlPort.java:28]: yes..

File: java/debugger.jpda.ui/src/org/netbeans/modules/debugger/jpda/ui/FixedWatchesManager.java (hunk 6 above)
Comment (BradWalker) [reply java/debugger.jpda.ui/src/org/netbeans/modules/debugger/jpda/ui/FixedWatchesManager.java:268]: yes..

**Comment by BradWalker** [comment]
Hey @mbien & @matthiasblaesing , could you take a quick look at this. I think I have everything up to do. But, it still fails a couple of the tests. It does not look like the test failures are related to me work though.

I need your eyes on it, if you don't mind.

**Review by mbien — APPROVED** [review]
looks good. restarted some tests and they are all green now.

**Pull request merged**